src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts

```diff
@@ -222,14 +222,29 @@ export function* watchCalculatePageOrderAndMoveToNextPageSaga(): SagaIterator {
 }
 
 export function* watchInitialCalculagePageOrderAndMoveToNextPageSaga(): SagaIterator {
-  yield all([
-    take(START_INITIAL_DATA_TASK_QUEUE_FULFILLED),
-    take(FormLayoutActions.fetchLayoutFulfilled),
-    take(FormLayoutActions.fetchLayoutSettingsFulfilled),
-  ]);
-  const state: IRuntimeState = yield select();
-  if (state.formLayout.uiConfig.layoutOrder.length > 1) {
-    yield put(FormLayoutActions.calculatePageOrderAndMoveToNextPage({ skipMoveToNext: true }));
+  while (true) {
```

Comment:
It isn't obvious to me how the logic escapes the loop.

Reply:
Yeah, it's not supposed to escape the loop :)

Reply:
Generator functions can be thought of like  pauseable functions, so putting while (true) in them does not mean blocking.